compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4374,6 +4377,64 @@ private STNode parseObjectTypeDescriptor() {
                 objectMembers, closeBrace);
     }
 
+    /**
+     * Parse object constructor expression.
+     *
+     * object-constructor-expr :=
+     *    [annots] [client] object [type-reference] {
+     *       object-member*
+     *    }
+     * object-member := object-field | method-defn
+     *
+     * @param isRhsExpr
+     * @param annots
+     * @return Parsed node
+     */
+    private STNode parseObjectConstructorExpressionNode(boolean isRhsExpr, STNode annots) {
+
+        startContext(ParserRuleContext.OBJECT_CONSTRUCTOR);
+
+        STNode objectTypeQualifier;
+        STToken nextToken = peek();
+        if (nextToken.kind == SyntaxKind.CLIENT_KEYWORD) {
+            startContext(ParserRuleContext.CLIENT_KEYWORD);
+            objectTypeQualifier = parseClientKeyword();
+            endContext();
+        } else {
+            objectTypeQualifier = STNodeFactory.createEmptyNode();
+        }
+
+        STNode objectKeyword = parseObjectKeyword();
+
+        STNode typeDescriptor;
+        nextToken = peek();
+        if (nextToken.kind == SyntaxKind.IDENTIFIER_TOKEN) {
+            typeDescriptor = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_NEW_EXPR);
+        } else {
+            typeDescriptor = STNodeFactory.createEmptyNode();
+        }
+
+        STNode objectCtorBody = parseObjectConstructorBody();
+
+        endContext();
+
+        return STNodeFactory.createObjectConstructorExpressionNode(annots,
+                objectTypeQualifier, objectKeyword, typeDescriptor, objectCtorBody);
+    }
+
+    private STNode parseObjectConstructorBody() {
+        startContext(ParserRuleContext.OBJECT_CONSTRUCTOR_BODY);
```

Comment:
don't need this context as well since we are already inside `OBJECT_CONSTRUCTOR` context